### PR TITLE
feat(plugin): add degradation switch for ext-plugin

### PR DESCRIPTION
**Pull request by arabot777**
### What this PR does / why we need it:

Add degradation switch for  ext-plugin.
https://github.com/apache/apisix/discussions/5833
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/zh/latest/plugins/ext-plugin-pre-req.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ External Plugins 执行的结果会影响当前请求的行为。
 | 名称      | 类型          | 必选项 | 默认值    | 有效值                                                                    | 描述                                                                                                                                         |
 | --------- | ------------- | ----------- | ---------- | ------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------- |
 | conf     | array        | 可选    |              | [{"name": "ext-plugin-A", "value": "{\"enable\":\"feature\"}"}] |     在 Plugin Runner 内执行的插件列表的配置 |
+| allow_degradation              | boolean  | 可选                                | false       |                                                                     | 当插件功能临时不可用时是否允许请求继续。当值设置为 true 时则自动允许请求继续，默认值是 false。|
```
Comment (tzssangglass) [inline docs/zh/latest/plugins/ext-plugin-pre-req.md:45]:
```suggestion
| allow_degradation              | boolean  | 可选                                | false       |                                                                     | 当 Plugin Runner 临时不可用时是否允许请求继续。当值设置为 true 时则自动允许请求继续，默认值是 false。|
```

File: docs/en/latest/plugins/ext-plugin-pre-req.md [hunk 2]
```diff
@@ -43,6 +43,7 @@ The result of external plugins execution will affect the behavior of the current
 | Name      | Type          | Requirement | Default    | Valid                                                                    | Description                                                                                                                                         |
 | --------- | ------------- | ----------- | ---------- | ------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------- |
 | conf     | array        | optional    |              | [{"name": "ext-plugin-A", "value": "{\"enable\":\"feature\"}"}] |     The plugins list which will be executed at the plugin runner with their configuration    |
+| allow_degradation              | boolean  | optional                                | false       |                                                                     | Whether to enable plugin degradation when the ext-plugin-pre-req function is temporarily unavailable. Allow requests to continue when the value is set to true, default false. |
```
Comment (spacewander) [inline docs/en/latest/plugins/ext-plugin-pre-req.md:46]:
```suggestion
| allow_degradation              | boolean  | optional                                | false       |                                                                     | Whether to enable plugin degradation when the plugin runner is temporarily unavailable. Allow requests to continue when the value is set to true, default false. |
```

**Review by spacewander — APPROVED** [review]

File: apisix/plugins/ext-plugin/init.lua [hunk 3]
```diff
@@ -700,6 +701,9 @@ function _M.communicate(conf, ctx, plugin_name)
 
         if not core.string.find(err, "conf token not found") then
             core.log.error(err)
+            if conf.allow_degradation then
+                return
```
Comment (tokers) [inline apisix/plugins/ext-plugin/init.lua:705]: Let's add a logging message here? So we can detect the degradation situation, also, we can enrich our test cases.

**Review by tzssangglass — APPROVED** [review]

File: apisix/plugins/ext-plugin/init.lua (hunk 3 above)
Comment (arabot777) [reply apisix/plugins/ext-plugin/init.lua:706]: have changed

**Review by tokers — APPROVED** [review]

**Review by tokers — CHANGES_REQUESTED** [review]
CI failed, should be fixed before merging.

**Comment by spacewander** [comment]
@tokers 
Can we merge this PR now?

**Review by tokers — APPROVED** [review]

**Pull request merged**